simple_parsing/annotation_utils/get_field_annotations.py

```diff
@@ -178,17 +177,22 @@ def get_field_type_from_annotations(some_class: type, field_name: str) -> type:
     # Get the local and global namespaces to pass to the `get_type_hints` function.
     local_ns: Dict[str, Any] = {"typing": typing, **vars(typing)}
     local_ns.update(forward_refs_to_types)
-    # Get the globals in the module where the class was defined.
-    global_ns = sys.modules[some_class.__module__].__dict__
-
+    # Get the global_ns in the module starting from the deepest base until the module where the field_name is defined.
+    global_ns = {}
+    for base_cls in reversed(some_class.mro()):
+        global_ns.update(sys.modules[base_cls.__module__].__dict__)
+        annotations = getattr(base_cls, '__annotations__', None)
```

Comment:
I'm surprised, I wouldn't have expected this to work! (since it stops loading the namespaces as soon as it finds a field with a matching name (e.g. as soon as it finds `overwrite_base.Foo`, in the example). :thinking:

Reply:
The loop invariant is that an annotation can only be defined at or above the scope it's used.